src/widget/wfindonmenu.cpp

```diff
@@ -0,0 +1,279 @@
+                this,
+                [this, service, trackProperty, albumName]() {
+                    emit triggerBrowser(service, trackProperty, albumName);
+                });
+        break;
+    }
+
+    case Service::Discogs: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, albumName),
+                this,
+                [this, service, trackProperty, albumName]() {
+                    emit triggerBrowser(service, trackProperty, albumName);
+                });
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+void WFindOnMenu::addActionsTrackTitle(Service service,
+        const QString& trackTitle,
+        QMenu* pServiceMenu,
+        TrackProperty trackProperty) {
+    const QString prefixAction = composePrefixAction(trackProperty);
+    switch (service) {
+    case Service::SoundCloud: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, trackTitle),
+                this,
+                [this, service, trackProperty, trackTitle]() {
+                    emit triggerBrowser(service, trackProperty, trackTitle);
+                });
+        break;
+    }
+
+    case Service::LastFm: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, trackTitle),
+                this,
+                [this, service, trackProperty, trackTitle]() {
+                    emit triggerBrowser(service, trackProperty, trackTitle);
+                });
+        break;
+    }
+
+    case Service::Discogs: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, trackTitle),
+                this,
+                [this, service, trackProperty, trackTitle]() {
+                    emit triggerBrowser(service, trackProperty, trackTitle);
+                });
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+void WFindOnMenu::openInBrowser(Service service,
+        TrackProperty trackProperty,
+        const QString& query) {
+    const QString serviceUrl = getServiceUrl(service, trackProperty);
+    QUrlQuery qurlquery;
+    qurlquery.addQueryItem("q", query);
+    QUrl url(serviceUrl);
+    url.setQuery(qurlquery);
+    QDesktopServices::openUrl(url);
+}
+
+void WFindOnMenu::populateWebLookUpQueries(QMenu* pServiceMenu,
+        const Track& track,
+        const QString& serviceTitle,
+        Service service) {
+    const auto artistTitle = track.getArtist();
+    pServiceMenu = new QMenu(this);
+    pServiceMenu->setTitle(serviceTitle);
+    addMenu(pServiceMenu);
+    addSeparator();
+    {
+        const auto trackTitle = track.getTitle();
+        const auto artistTitleWithTitle = composeSearchQuery(trackTitle, artistTitle);
+        if (!trackTitle.isEmpty()) {
+            addActionsTrackTitle(service,
+                    trackTitle,
+                    pServiceMenu,
+                    WFindOnMenu::TrackProperty::Title);
+            addActionsTrackTitle(service,
```

Comment:
addActionsTrackTitle() that is invoked with a parameter ArtistTitle? Sorry, you lost me.

Reply:
Sorry. Same as before, all ArtistTitle's are going to be "primaryArtist".